### ISSUE TITLE
{"message": "useragent mismatch", "status": "fail"} UploadPhoto not working

### ISSUE THREAD

**Issue by RHOMBI** [mgp25/Instagram-API#3077]
# Bug Report

---

### Notes:
We will close your issue, sometimes without answering, if any of the following are met:
* You have not included your code or your debug log
* You are asking about `challenge_required`, `checkpoint_required`, `feedback_required` or `sentry_block`. They've already been answered in the Wiki and *countless* closed tickets in the past!
* You have used the wrong issue template
* You are posting screenshots, which are too painful to help you with 

Please make sure you have done the following before submitting your issue:
* You have [Searched](https://github.com/mgp25/Instagram-API/search?type=Issues) the bug-tracker for similar issues including **closed** ones
* You have [Read the FAQ](https://github.com/mgp25/Instagram-API/wiki/FAQ)
* You have [Read the wiki](https://github.com/mgp25/Instagram-API/wiki)
* You have [Reviewed the examples](https://github.com/mgp25/Instagram-API/tree/master/examples)
* You have [Installed the api using ``composer``](https://github.com/mgp25/Instagram-API#installation)
* You are [Using latest API release](https://github.com/mgp25/Instagram-API/releases)

---

### Description
Please include a well-worded description of your issue below; We will close your issue if we have to guess what you're talking about, please be as specific as possible:

__INSERT YOUR DESCRIPTION HERE__

### Code
Please post your code relevant to the bug in the section below:
```php
<?php

set_time_limit(0);
date_default_timezone_set('UTC');

require __DIR__.'/../vendor/autoload.php';

/////// CONFIG ///////
//$userId = xxx;
$username = 'xxx';
$password = 'xxx';
$debug = true;
$truncatedDebug = false;
//////////////////////

/////// MEDIA ////////
$photoFilename = 'sad700.jpeg';
$captionText = 'fire';
//////////////////////

$ig = new \InstagramAPI\Instagram($debug, $truncatedDebug);

try {
    $ig->login($username, $password);
} catch (\Exception $e) {
    echo 'Something went wrong: '.$e->getMessage()."\n";
    exit(0);
}
sleep(3);
echo('well that worked');
#try {
    // The most basic upload command, if you're sure that your photo file is
    // valid on Instagram (that it fits all requirements), is the following:
    // $ig->timeline->uploadPhoto($photoFilename, ['caption' => $captionText]);

    // However, if you want to guarantee that the file is valid (correct format,
    // width, height and aspect ratio), then you can run it through our
    // automatic photo processing class. It is pretty fast, and only does any
    // work when the input file is invalid, so you may want to always use it.
    // You have nothing to worry about, since the class uses temporary files if
    // the input needs processing, and it never overwrites your original file.
    //
    // Also note that it has lots of options, so read its class documentation!
    $photo = new \InstagramAPI\Media\Photo\InstagramPhoto($photoFilename);
    echo('work2');
    $ig->timeline->uploadPhoto($photoFilename, ['caption' => $captionText]);
    echo('work3');
#} catch (\Exception $e) {
    #echo 'Something went wrong: '.$e->getMessage()."\n";
#}

``` 

### Debug Log
Please post your debug log in the section below; You can enable the debug log by using `new Instagram(true)` instead of `new Instagram()`. Additionally, you can do `\InstagramAPI\Debug::$debugLog = true;` for the debug log to be saved to a file:
```php
C:\xampp\htdocs\Instagram-API-master\examples>php uploadPhoto.php
well that workedwork2POST:  https://i.instagram.com/api/v1/upload/photo/
→ 44.53kB
← 400    58B
RESPONSE: Something is wrong. Please check your input and try again.

PHP Fatal error:  Uncaught InstagramAPI\Exception\BadRequestException: Invalid request options. in C:\xampp\htdocs\Instagram-API-master\src\Client.php:499
Stack trace:
#0 C:\xampp\htdocs\Instagram-API-master\src\Request.php(834): InstagramAPI\Client->mapServerResponse()
#1 C:\xampp\htdocs\Instagram-API-master\src\Request\Internal.php(1740): InstagramAPI\Request->getResponse()
#2 C:\xampp\htdocs\Instagram-API-master\src\Request\Internal.php(139): InstagramAPI\Request\Internal->_uploadPhotoInOnePiece()
#3 C:\xampp\htdocs\Instagram-API-master\src\Request\Internal.php(101): InstagramAPI\Request\Internal->uploadPhotoData()
#4 C:\xampp\htdocs\Instagram-API-master\src\Request\Timeline.php(38): InstagramAPI\Request\Internal->uploadSinglePhoto()
#5 C:\xampp\htdocs\Instagram-API-master\examples\uploadPhoto.php(46): InstagramAPI\Request\Timeline->uploadPhoto()
#6 {main}
  thrown in C:\xampp\htdocs\Instagram-API-master\src\Client.php on line 499

```

When i click the link given in the error via browser, i see :
{"message": "useragent mismatch", "status": "fail"}

**Comment by JRoy** [comment]
Duplicate of #2972

**Comment by shadowgroundz** [comment]
@JRoy duplicate? So whats solution? No solution everywhere. Post always deleted!!!!!!

**Comment by JRoy** [comment]
It doesn't matter if there's no solution, it's still a duplicate. You don't open a new ticket just before the last one doesn't have a solution

**Comment by Julian-O** [comment]
@JRoy: I understand the desire to keep the conversation in one place, however #2971 and #2972 are both locked and silent without explanation. #2971 *appears* like after monitoring for a while, a decision was made that there was no problem, and it was closed. Further, there was limited information about the problem in #2971, so this bug report contained more useful info.

Can I politely ask that you post some sort of information in #2971 and/or the Wiki about this issue? e.g. Do the maintainers acknowledge that it is still a problem (because it is!)? Do you know the cause? Should we expect a solution and/or are their ways we can contribute to a solution?

I know that other API's for Python are suffering the same problem. Users of instapy-cli are resorting to [rather hacky Selenium/autoit workarounds](https://github.com/instagrambot/instapy-cli/issues/75) and I would rather use a cleaner solution.
